osquery/tables/system/darwin/diagnostics.cpp

```diff
@@ -0,0 +1,133 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed in accordance with the terms specified in
+ *  the LICENSE file found in the root directory of this source tree.
+ */
+
+#include <ctime>
+#include <sstream>
+#include <string>
+#include <vector>
+
+#include <boost/algorithm/string/replace.hpp>
+#include <boost/property_tree/json_parser.hpp>
+#include <boost/property_tree/ptree.hpp>
+
+#include <osquery/filesystem/filesystem.h>
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+#include <osquery/utils/conversions/split.h>
+
+namespace pt = boost::property_tree;
+
+namespace osquery {
+namespace tables {
+
+// read data from core_analytics file
+void genDiagnosticsFromFile(const boost::filesystem::path& diagnostics_file,
+                            QueryData& results) {
+  std::string diagnostics_content;
+
+  if (forensicReadFile(diagnostics_file, diagnostics_content).ok()) {
+    std::string diag_start;
+    std::string diag_end;
+
+    for (const auto& line : split(diagnostics_content, "\n")) {
+      if (line.find("startTimestamp") != std::string::npos) {
+        pt::ptree tree;
+        std::stringstream ss(line);
+
+        pt::read_json(ss, tree);
+        diag_start = tree.get<std::string>("startTimestamp");
+
+        // remove letters in timestamp
+        boost::replace_all(diag_start, "T", " ");
+        boost::replace_all(diag_start, "Z", "");
```

Comment:
I am not quite sure on this, but I don't think we can just remove these characters. IIRC the 'Z' is Zulu time (UTC 0). I think this is represented in 8601 format, and we probably need to convert it to unix time.

Reply:
Agreed. This timestamp is in rfc3339 format. I'd say either use it as is, or pull in a parser for it.